Trigger command module

HubbeKing opened the pull request:
As per Issue #5 , I thought we might do a Trigger module?

HubbeKing committed:
Basic skeleton for Trigger command module

HubbeKing committed:
Improved subcommand-based skeleton (thanks Alias!)

HubbeKing committed:
Make _delTrigger

HubbeKing committed:
Slightly more useful help-stub

HubbeKing committed:
Make _addTrigger

HubbeKing committed:
Make import and export functions, add lots of docstrings

HubbeKing commented:
Mostly done now, just need to make a not-useless help() for the module

StarlitGhost reviewed (commented):
Looking good so far 👍

HubbeKing committed:
Flesh out help?

HubbeKing committed:
Actual proper help(), probably

HubbeKing committed:
Add some basic tests for Trigger

HubbeKing committed:
Correct misspelling of parameters

HubbeKing committed:
Finally working help?

HubbeKing committed:
subCommands, not subcommands. camelCase!

HubbeKing committed:
Fix docstrings to hopefully work with help()

HubbeKing committed:
Extend BotModule rather than BotCommand

Should be fine, since self.execute is being fired on every message-channel/message-user/action-channel/action-user anyway.

HubbeKing committed:
More test commands for Trigger

HubbeKing committed:
Extend BotCommand to get @admin working

But use ownTriggers rather than triggers() to prevent multiple actions from firing from one !trigger message

HubbeKing committed:
The dict is triggerData, not triggerName, me.

HubbeKing committed:
Whoops, how'd that get pasted there

HubbeKing committed:
You need .items() to iterate over a dict's items

HubbeKing committed:
Update test_commands.txt

HubbeKing committed:
Add debug logger to _handleTriggerCommand

HubbeKing committed:
Fix trigger adding

HubbeKing committed:
Fix trigger handling

HubbeKing committed:
Fix Trigger example

StarlitGhost committed:
switch to dbco for trigger exports
